cpp/ycm/ClangCompleter/TranslationUnit.cpp

```diff
@@ -452,6 +452,41 @@ std::vector< FixIt > TranslationUnit::GetFixItsForLocationInFile(
   return fixits;
 }
 
+DocumentationData TranslationUnit::GetDocsForLocationInFile(
+  int line,
+  int column,
+  const std::vector< UnsavedFile > &unsaved_files,
+  bool reparse ) {
+
+  if ( reparse )
+    ReparseForIndexing( unsaved_files );
+
+  unique_lock< mutex > lock( clang_access_mutex_ );
+
+  if ( !clang_translation_unit_ )
+    return DocumentationData();
+
+  CXCursor cursor = GetCursor( line, column );
+
+  if ( !CursorIsValid( cursor ) )
+    return DocumentationData();
+
+  // If the original cursor is a reference, then we return the documentation
+  // for the type/method/etc. that is referenced
+  CXCursor referenced_cursor = clang_getCursorReferenced( cursor );
+  if ( CursorIsValid( referenced_cursor ) ) {
+    cursor = referenced_cursor;
+  }
```

Comment:
I don't care do you use braces around single-line blocks or not except that you keep it consistent inside a function. :)

Otherwise it's bedlam and Cthulhu.